client/src/logic/icsGen.ts

```diff
@@ -12,22 +12,31 @@ export interface Course {
   assessments: Assessment[];
 }
 
-function jsonToICS(semester: Course[]): string {
+export interface CourseData {
+  topic: string;
+  assessments: Assessment[];
+}
+
+export interface Response {
```

Comment:
Please rename class names as they're quite confusing. And `Course` seems to not be used anywhere.

Consider following changes:
- `Course` -> `CourseInfo`
- `CourseData` -> `Course`, and the class contains `info: CourseInfo` and `assessments: Assessment[]`; **or** only `Course` which contains: `topic: string`, `description: string`, `hours: string`, etc and `assessments: Assessment[]`.
- `Response` -> `Courses`

If you wish to have a class `Response` for http requests, consider having a wrapping class like:
```
interface Response {
    statusCode: int;
    courses?: Courses;
    error?: string;
    ....
}
```